buildpacks/scala/src/layers/coursier_cache.rs

```diff
@@ -0,0 +1,99 @@
+use crate::ScalaBuildpack;
+use libcnb::build::BuildContext;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::generic::GenericMetadata;
+use libcnb::layer::{ExistingLayerStrategy, Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::layer_env::{LayerEnv, ModificationBehavior, Scope};
+use libcnb::Buildpack;
+use std::path::Path;
+
+pub struct CoursierCacheLayer {
+    pub available_at_launch: Option<bool>,
+}
+
+// Coursier is used instead of Ivy for library management starting with sbt >= 1.3
+// https://www.scala-sbt.org/1.x/docs/sbt-1.3-Release-Notes.html
+impl Layer for CoursierCacheLayer {
+    type Buildpack = ScalaBuildpack;
+    type Metadata = GenericMetadata;
+
+    fn types(&self) -> LayerTypes {
+        LayerTypes {
+            build: true,
+            launch: self.available_at_launch.unwrap_or_default(),
+            cache: true,
+        }
+    }
+
+    fn create(
+        &self,
+        _: &BuildContext<Self::Buildpack>,
+        layer_path: &Path,
+    ) -> Result<LayerResult<Self::Metadata>, <Self::Buildpack as Buildpack>::Error> {
+        LayerResultBuilder::new(GenericMetadata::default())
+            .env(create_coursier_layer_env(
+                layer_path,
+                self.available_at_launch,
+            ))
+            .build()
+    }
+
+    fn existing_layer_strategy(
+        &self,
+        _: &BuildContext<Self::Buildpack>,
+        _: &LayerData<Self::Metadata>,
+    ) -> Result<ExistingLayerStrategy, <Self::Buildpack as Buildpack>::Error> {
+        Ok(ExistingLayerStrategy::Keep)
```

Comment:
For the user, both logs are kinda useless since this is basically just `mkdir -p`, nothing else happens here. I agree that it is useful for maintainers to be able to see the layer lifecycles. Together with your comment asking @colincasey to add this to _every layer_, it seems like a concern we should tackle outside of the actual buildpacks. Maybe a logging flag that tells libcnb itself to log the layer lifecycles (and potentially more stuff)?